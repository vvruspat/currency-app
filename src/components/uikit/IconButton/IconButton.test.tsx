import { render, screen } from "@testing-library/react";
import { IconButton } from ".";

test("renders IconButton without icon", () => {
  render(
    <IconButton data-testid="test-icon-button">Test Button</IconButton>,
    {}
  );
  const IconButtonElement = screen.getByTestId("test-icon-button");
  expect(IconButtonElement).toBeInTheDocument();
  expect(IconButtonElement).toHaveTextContent("Test Button");
  expect(IconButtonElement).toMatchSnapshot();
});

test("renders IconButton with icon", () => {
  render(
    <IconButton
      data-testid="test-icon-button"
      icon={<div data-testid="test-icon-button-icon"></div>}
    >
      Test Button
    </IconButton>,
    {}
  );
  const IconButtonIconElement = screen.getByTestId("test-icon-button-icon");
  expect(IconButtonIconElement).toBeInTheDocument();

  const IconButtonElement = screen.getByTestId("test-icon-button");
  expect(IconButtonElement).toMatchSnapshot();
});
