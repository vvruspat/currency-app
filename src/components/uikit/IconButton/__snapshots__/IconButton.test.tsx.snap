// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders IconButton with icon 1`] = `
<button
  class="icon-button"
  data-testid="test-icon-button"
>
  <div
    class="icon-button-content"
  >
    <div
      class="icon-button-icon"
    >
      <div
        data-testid="test-icon-button-icon"
      />
    </div>
    <div
      class="icon-button-text"
    >
      Test Button
    </div>
  </div>
</button>
`;

exports[`renders IconButton without icon 1`] = `
<button
  class="icon-button"
  data-testid="test-icon-button"
>
  <div
    class="icon-button-content"
  >
    <div
      class="icon-button-icon"
    />
    <div
      class="icon-button-text"
    >
      Test Button
    </div>
  </div>
</button>
`;
